TODO.md

```diff
@@ -0,0 +1,69 @@
+# TODO
+remaining tasks for the `0.2` release, in order of priority
+
+## major
+* [ ] move behaviours (getting, updating, etc.) to library methods
+* [ ] validation (schema, column, relationship)
+* [ ] declarative entity definitions
+* [ ] schema validation
+* [ ] documentation
+* [ ] callbacks
+* [ ] pagination
+* [ ] hateoas-style support
+
+### move behaviours
+at the moment, when you generate your code, a lot of functionality
+gets put in the restplus endpoint. this results in huge diffs between
+versions and is tiresome to test.
+
+it is proposed that this code moves into a library function which then
+gets imported by the endpoint. this will make the code easier to test,
+and will result in a smaller generated codebase.
+
+### validation
+there's loads that could be done to validate a schema. the most interesting
+part is probably the relationships - to do this we would need a two-pass
+implementation where the first pass builds up an idea of what entities are
+present and how they relate to one another, and the second pass
+verifies that these relationships are possible.
+
+error reporting should be as approachable and helpful as possible. see
+[elm](https://elm-lang.org/) for inspiration.
+
+### declarative entity definitions
+writing entities declaratively is much more natural to python developers
+already familiar with sqlalchemy, which is a dependency of genyrator.
+it is proposed to use decorators in the style of
+[attr.s](http://www.attrs.org/en/stable/) to configure the behaviour of
+entities.
+
+### schema validation
```

Comment:
I would strongly encourage against implementing yet another validation engine. There are two (three if you count jsonschema) as part of genyrator already (marshmallow and flask-resplus). I think it would be better if figured out how to lean on one of those more.

Given that flask-restplus seems to be unmaintained now and marshmallow is widely used in the community my preference would be there. Even if it is a bit lacking.

Reply:
agreed, would be nice if we could do this with an existing framework :+1:

Reply:
marshmallow-sqlalchemy has support for [overriding fields](https://marshmallow-sqlalchemy.readthedocs.io/en/latest/recipes.html#overriding-generated-fields) and for [manually generating fields](https://marshmallow-sqlalchemy.readthedocs.io/en/latest/recipes.html#overriding-generated-fields) (that function accepts keyword arguments that are passed on down to the marshmallow field) so this does not sound very difficult.